include/filewriter/summarywriter.hpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * @file filewriter.hpp
+ * @author Jeffin Sam
+ */
+#ifndef MLBOARD_SUMMARY_WRITER_HPP
+#define MLBOARD_SUMMARY_WRITER_HPP
+
+#include <../core.hpp"
```

Comment:
I am getting error on this line

Reply:
I fixed it. I am now getting error while  `mlboard_tests`
```
CMakeFiles/mlboard_tests.dir/main.cpp.o: In function `mlboard::FileWriter::WriteSummary()':
main.cpp:(.text+0x774): undefined reference to `google::protobuf::MessageLite::SerializeToString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const'
main.cpp:(.text+0x897): undefined reference to `mlboard::Event::~Event()'
main.cpp:(.text+0x935): undefined reference to `mlboard::Event::~Event()'
CMakeFiles/mlboard_tests.dir/main.cpp.o: In function `mlboard::FileWriter::CreateEvent(unsigned long, mlboard::Summary*)':
main.cpp:(.text+0x982): undefined reference to `mlboard::Event::Event()'
main.cpp:(.text+0x9d5): undefined reference to `mlboard::Event::set_allocated_summary(mlboard::Summary*)'
main.cpp:(.text+0x9f4): undefined reference to `mlboard::Event::~Event()'
main.cpp:(.text+0xa15): undefined reference to `mlboard::Event::~Event()'
CMakeFiles/mlboard_tests.dir/main.cpp.o: In function `std::thread::thread<void (mlboard::FileWriter::*)(), mlboard::FileWriter*>(void (mlboard::FileWriter::*&&)(), mlboard::FileWriter*&&)':
main.cpp:(.text._ZNSt6threadC2IMN7mlboard10FileWriterEFvvEJPS2_EEEOT_DpOT0_[_ZNSt6threadC5IMN7mlboard10FileWriterEFvvEJPS2_EEEOT_DpOT0_]+0x33): undefined reference to `pthread_create'
CMakeFiles/mlboard_tests.dir/main.cpp.o: In function `mlboard::SharedQueue<mlboard::Event>::pop()':
main.cpp:(.text._ZN7mlboard11SharedQueueINS_5EventEE3popEv[_ZN7mlboard11SharedQueueINS_5EventEE3popEv]+0x7a): undefined reference to `mlboard::Event::Event(mlboard::Event const&)'
main.cpp:(.text._ZN7mlboard11SharedQueueINS_5EventEE3popEv[_ZN7mlboard11SharedQueueINS_5EventEE3popEv]+0xd6): undefined reference to `mlboard::Event::~Event()'
CMakeFiles/mlboard_tests.dir/main.cpp.o: In function `void __gnu_cxx::new_allocator<mlboard::Event>::construct<mlboard::Event, mlboard::Event const&>(mlboard::Event*, mlboard::Event const&)':
main.cpp:(.text._ZN9__gnu_cxx13new_allocatorIN7mlboard5EventEE9constructIS2_JRKS2_EEEvPT_DpOT0_[_ZN9__gnu_cxx13new_allocatorIN7mlboard5EventEE9constructIS2_JRKS2_EEEvPT_DpOT0_]+0x48): undefined reference to `mlboard::Event::Event(mlboard::Event const&)'
collect2: error: ld returned 1 exit status
tests/CMakeFiles/mlboard_tests.dir/build.make:98: recipe for target 'mlboard_tests' failed
make[2]: *** [mlboard_tests] Error 1
CMakeFiles/Makefile2:145: recipe for target 'tests/CMakeFiles/mlboard_tests.dir/all' failed
make[1]: *** [tests/CMakeFiles/mlboard_tests.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2

```

Reply:
This linking error is most probably because `-lproto -lprotobuf` is not present while compiling. As the error message looks familiar.
So, I think it's a compatibility issue.

Reply:
I tried every possible way out, but I wasn't able to get port_def.inc in my include dir